### DISCUSSION TITLE
How to build, test and run

### DISCUSSION THREAD

**Discussion by winksaville** [dahliaOS/terminal#8] [General]
As a dahliaOS neophyte I need to know how to build the terminal, test and then run it on my "device". Hopefully there are some general instructions on how to get started, please point me to those. But I suspect there are particulars for the terminal, please provide any hints.

I'll gladly update the "README.md" or some other file, as you as you see fit, so the next person will just see the instructions.

**Comment by imreallyliam** [comment]
I'll be adding instructions to the README in the next few commits but overall it's pretty simple. However, for debugging purposes, the terminal is set to use `cmd` rather than dynamically selection based on the operating system, so before doing the following make sure that you've modified line 23 in lib/FrameBackend.dart to use the shell you'd like. 

1. (Windows, macOS [untested], Linux) Install Flutter and enable desktop support for your operating system. This generally follows `flutter config --enable-[platform]-desktop`.
2. Run `flutter doctor -v` for your platform and install any dependencies.
3. In the project, run `flutter pub get` then to run `flutter run -d [platform]`

In theory, the system will work on macOS, Windows, and Linux once completed, minus a few platform-specific changes. There isn't a Zircon build yet as we don't have a dahliaOS build for it quite yet, but this is also an area where help would be greatly appreciated.
